Wrappers/Python/ccpi/web_viewer/trame_viewer3D.py

```diff
@@ -458,3 +458,45 @@ def show_slice_histogram(self, show_histogram):
         else:
             self.cil_viewer.histogramPlotActor.VisibilityOff()
         self.html_view.update()
+
+    def change_slice_number(self, slice_number):
+        self.cil_viewer.updateSliceHistogram()
+        self.cil_viewer.setActiveSlice(slice_number)
+        self.cil_viewer.updatePipeline()
+        self.html_view.update()
+
+    def change_slice_level(self, new_level: float, current_window: float = None):
+        if self.slice_window_slider_is_percentage:
+            if current_window is None:
+                current_window = new_level
+            self.cil_viewer.setSliceColorPercentiles(current_window, new_level)
+        else:
+            self.cil_viewer.setSliceColorLevel(level=new_level)
+        self.cil_viewer.updatePipeline()
+        self.html_view.update()
+
+    def change_slice_window(self, new_window: float, current_level: float = None):
+        if self.slice_window_slider_is_percentage:
+            if current_level is None:
+                current_level = new_window
+            self.cil_viewer.setSliceColorPercentiles(new_window, current_level)
+        else:
+            self.cil_viewer.setSliceColorWindow(window=new_window)
+        self.cil_viewer.updatePipeline()
+        self.html_view.update()
+
+    def change_slice_window_range(self, window: float, level: float):
+        if self.slice_window_slider_is_percentage:
+            min_percentage = level - window / 2
+            max_percentage = window + level - window / 2
+            self.cil_viewer.setSliceColorPercentiles(min_percentage, max_percentage)
+        else:
+            self.cil_viewer.setSliceColorWindowLevel(window, level)
+        self.cil_viewer.updatePipeline()
+        self.html_view.update()
```

Comment:
why is this percentage stuff a thing only in the 3D case?

Reply:
Oh I think the percentiles method only exists in the 3D viewer currently? So another thing for me to update